frontends/main/src/app-pages/B2BAttachPage/B2BAttachPage.tsx

```diff
@@ -23,15 +26,37 @@ const B2BAttachPage: React.FC<B2BAttachPageProps> = ({ code }) => {
     enrollment_code: code,
   })
 
+  const { isLoading: userLoading, data: user } = useQuery({
+    ...userQueries.me(),
+    staleTime: 0,
+  })
+  const { data: mitxOnlineUser } = useMitxOnlineCurrentUser()
+
   React.useEffect(() => {
     attach?.()
   }, [attach])
 
   React.useEffect(() => {
+    if (userLoading) {
+      return
+    }
+    if (!user?.is_authenticated) {
+      const loginUrlString = urls.login({
+        pathname: urls.b2bAttachView(code),
+        searchParams: new URLSearchParams(),
+      })
+      const loginUrl = new URL(loginUrlString)
+      const skipOnboarding = new URLSearchParams({ skip_onboarding: "1" })
+      loginUrl.search = skipOnboarding.toString()
```

Comment:
You don't want `loginUrl.search = ...`, that overrides the existing search params, which include the `next`. parameter. So
```ts
// this turns
"/login?next=/attach/uuid"
// into
"login?skip_onboarding=1"
```

You want:
```
loginUrl.searchParams.set("skip_onboarding", "1")
```